core/tee/tee_rpmb_fs.c

```diff
@@ -2369,19 +2372,87 @@ static TEE_Result rpmb_fs_read(struct tee_file_handle *tfh, size_t pos,
 	return res;
 }
 
+static TEE_Result update_write_helper(struct rpmb_file_handle *fh,
+				      size_t pos, const void *buf,
+				      size_t size, uintptr_t new_fat,
+				      size_t new_size)
+{
+	uintptr_t old_fat = fh->fat_entry.start_address;
+	size_t old_size = fh->fat_entry.data_size;
+	const uint8_t *rem_buf = buf;
+	size_t rem_size = size;
+	uint8_t *blk_buf = NULL;
+	size_t blk_offset = 0;
+	size_t blk_size = 0;
+	TEE_Result res = TEE_ERROR_GENERIC;
+	size_t rpmb_bytes_to_write = new_size;
+
+	if (!rpmb_bytes_to_write)
+		return TEE_SUCCESS;
+
+	blk_buf = mempool_alloc(mempool_default, TMP_BLOCK_SIZE);
+	if (!blk_buf)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	while (rpmb_bytes_to_write) {
```

Comment:
I'd prefer dropping `rpmb_bytes_to_write` and test `blk_offset < new_size` instead.

Reply:
makes sense. ok.